### ISSUE TITLE
False positive from UnusedImports on org.gradle.kotlin.dsl.provideDelegate

### ISSUE THREAD

**Issue by alphaho** [arturbosch/detekt#1608]
Hi,

I've found that when I'm writing the gradle buildSrc using kotlin dsl, and use the kotlin delegation, I would need to import `org.gradle.kotlin.dsl.provideDelegate`. But Detekt would give me an `UnusedImports` error in this case.

For example, with the following code, detekt task would fail:

```kotlin
import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.tasks.WriteProperties
import org.gradle.kotlin.dsl.getValue
import org.gradle.kotlin.dsl.provideDelegate  // got UnusedImports from this line
import org.gradle.kotlin.dsl.registering

class DumpVersionProperties : Plugin<Project> {
    override fun apply(target: Project) {
        with(target) {
            val dumpVersionProperties by tasks.registering(WriteProperties::class) {
                setProperties(mapOf("version" to "1.2.3"))
                outputFile = rootDir.resolve("version.properties")
            }

        }
    }
}
```

## Expected Behavior
It should not report `UnusedImports` if I do use it for delegation.

## Observed Behavior
Detekt task fails and report there's an `UnusedImports` error.

## Your Environment

* Version of detekt used: 1.0.0-RC14
* Version of Gradle used (if applicable): 5.4
* Operating System and version: Windows 10

**Comment by 3flex** [comment]
This is interesting, because `provideDelegate` is not used in the file, but still has to be imported otherwise build fails. I feel that detekt's behaviour is correct and it's only flagging this because of an issue either in Gradle Kotlin DSL or Kotlin itself, because I don't know expect this implementation detail to be exposed here.

See KT-22101 and https://github.com/Gradle/kotlin-dsl/issues/564

Not sure what best thing to do is, since trying to whitelist this case in detekt doesn't seem the right approach.

**Comment by alphaho** [comment]
Hmm, that makes sense to me.

But looking at the [delegated properties doc](https://kotlinlang.org/docs/reference/delegated-properties.html#providing-a-delegate-since-11), this imported function seems to be an overloaded operator. So when we use the delegated property in scope, it should be considered as being in used theoretically. 
But not sure whether we can tell that difference or not from Detekt's perspective.

**Comment by 3flex** [comment]
Sorry, haven't had time to look at this more closely.

Can you see what ktlint does here? It has a similar rule.

**Comment by 3flex** [comment]
Couldn't help myself - there's a whitelist of all the operators in the rule in detekt, just need to add `provideDelegate` to it. I'll submit a PR shortly unless someone beats me to it!

**Comment by alphaho** [comment]
Sounds good to me

**Comment by alphaho** [comment]
have tried running ktlint, it would also report `provideDelegate` as Unused import
